src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+            CancellationToken cancellationToken)
+        {
+            var firstCallSite = callSites.Keys.First();
+            var modifiedSolution = firstCallSite.Project.Solution;
+            var newMethodIdentifier = methodSymbol.Name + "_" + parameterName;
+
+            foreach (var grouping in callSites.GroupBy(kvp => kvp.Key.Project))
+            {
+                var project = grouping.Key;
+                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                foreach (var (document, invocationExpressionList) in grouping)
+                {
+                    var generator = SyntaxGenerator.GetGenerator(document);
+                    var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+                    var invocationSemanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                    var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var editor = new SyntaxEditor(root, generator);
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                    var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), ascendOutOfTrivia: true)!;
+                    var insertionIndex = GetInsertionIndex(semanticDocument, compilation, syntaxFacts, oldMethodDeclaration, cancellationToken);
+
+                    if (trampoline)
+                    {
+                        foreach (var invocationExpression in invocationExpressionList)
+                        {
+                            editor.ReplaceNode(invocationExpression, (currentInvocation, _) =>
+                            {
+                                return GenerateNewInvocationExpressionForTrampoline(syntaxFacts, editor, currentInvocation, invocationExpression, newMethodIdentifier, insertionIndex);
+                            });
+                        }
+                    }
+
+                    if (document.Id == semanticDocument.Document.Id)
+                    {
+                        if (trampoline)
+                        {
+                            GenerateNewMethod(semanticDocument, expression, methodSymbol, newMethodIdentifier, editor, cancellationToken);
+                        }
+                        else
+                        {
+                            GenerateNewMethodOverload(semanticDocument, expression, methodSymbol, editor);
+                        }
+
+                        UpdateExpressionInOriginalFunction(semanticDocument, expression, oldMethodDeclaration, parameterName, editor, allOccurrences, cancellationToken);
+
+                        var parameterType = semanticDocument.SemanticModel.GetTypeInfo(expression, cancellationToken).Type ?? semanticDocument.SemanticModel.Compilation.ObjectType;
```

Comment:
you don't do the `??` in that other piece of code i saw that was generating the method return type.

Reply:
bc the CodeGenerationSymbolFactory reverts to the MethodSymbol type, but I will have it use ObjectType in case the method type is void.